lighthouse-core/computed/main-thread-tasks.js

```diff
@@ -177,27 +173,35 @@ class MainThreadTasks {
        */
       case 'v8.compile':
       case 'EvaluateScript':
+        if (task.event.args.data) {
+          taskURLs = [task.event.args.data.url].concat(
+            (task.event.args.data.stackTrace || []).map(entry => entry.url));
+        }
+        break;
       case 'FunctionCall':
-        taskURLs = [argsData.url].concat(stackFrameURLs);
+        taskURLs = [task.event.args.data.url];
         break;
       case 'v8.compileModule':
-        taskURLs = [task.event.args.fileName].concat(stackFrameURLs);
+        taskURLs = [task.event.args.fileName];
+        if (task.event.args.data) {
+          taskURLs = taskURLs.concat(
+            task.event.args.data.stackTrace.map(entry => entry.url));
+        }
         break;
       case 'TimerFire': {
-        /** @type {string} */
-        // @ts-ignore - timerId exists when name is TimerFire
         const timerId = task.event.args.data.timerId;
         const timerInstallerTaskNode = priorTaskData.timers.get(timerId);
         if (!timerInstallerTaskNode) break;
-        taskURLs = timerInstallerTaskNode.attributableURLs.concat(stackFrameURLs);
+        taskURLs = timerInstallerTaskNode.attributableURLs;
```

Comment:
notice I removed access to `stackFrame` - I don't think it exists on `TimerFire`.

Reply:
correct. it doesn't. https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/inspector/inspector_trace_events.cc?l=878-885&rcl=a7ed373a087d07f91d9a58c03da1739d48e7f7ea